joern-cli/frontends/csharpsrc2cpg/src/main/scala/io/joern/csharpsrc2cpg/datastructures/CSharpProgramSummary.scala

```diff
@@ -51,6 +55,30 @@ object CSharpProgramSummary {
   def jsonToInitialMapping(jsonInputStream: InputStream): Try[NamespaceToTypeMap] =
     Try(read[NamespaceToTypeMap](ujson.Readable.fromByteArray(jsonInputStream.readAllBytes())))
 
+  def mergeBuiltInTypesJson: InputStream = {
+    val classLoader      = getClass.getClassLoader
+    val builtinDirectory = "builtin"
+    val resourcePaths = Option(classLoader.getResources(s"$builtinDirectory/"))
+      .map(_.asScala.map(_.getPath.stripPrefix("/")).toList)
+      .getOrElse(List.empty[String])
+      .filter(_.endsWith("builtin_types.json"))
+
+    if (resourcePaths.isEmpty) {
+      logger.warn("No builtin_types.json found.")
+      InputStream.nullInputStream()
+    } else {
+      val mergedJsonObjects = ListBuffer[ujson.Obj]()
+      for (resourcePath <- resourcePaths) {
+        val inputStream = classLoader.getResourceAsStream(resourcePath)
+        val jsonString  = Source.fromInputStream(inputStream).mkString
+        val jsonObject  = ujson.read(jsonString).obj
+        mergedJsonObjects += jsonObject
+      }
+      val mergedJson = write(mergedJsonObjects.map(_.value).reduce(_ ++ _))
```

Comment:
We may want to go for a `reduceOption` in case `mergedJsonObjects` remains empty